tests/serialized_test_data_generation/serialbox_to_netcdf.py

```diff
@@ -0,0 +1,134 @@
+import sys
+import argparse
+import os
+import shutil
+import warnings
+import xarray as xr
+import f90nml
+import numpy as np
+sys.path.append("/serialbox/python")  # noqa: E402
+import serialbox  # noqa: E402
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        "converts serialbox data to netcdf"
+    )
+    parser.add_argument(
+        "data_path", type=str, help="path of serialbox data to convert",
+    )
+    parser.add_argument(
+        "output_path", type=str, help="output directory where netcdf data will be saved"
+    )
+    return parser
+
+
+def read_serialized_data(serializer, savepoint, variable):
+    data = serializer.read(variable, savepoint)
+    if len(data.flatten()) == 1:
+        return data[0]
+    return data
+
+
+def get_all_savepoint_names(data_path):
+    savepoint_names = set()
+    serializer = get_serializer(data_path, rank=0)
+    for savepoint in serializer.savepoint_list():
+        savepoint_names.add(savepoint.name)
+    return savepoint_names
+
+
+def get_serializer(data_path, rank):
+    return serialbox.Serializer(
+        serialbox.OpenModeKind.Read, data_path, "Generator_rank" + str(rank)
+    )
+
+
+def main(data_path: str, output_path: str):
+    os.makedirs(output_path, exist_ok=True)
+    namelist_filename_in = os.path.join(data_path, "input.nml")
+    namelist_filename_out = os.path.join(output_path, "input.nml")
+    if namelist_filename_out != namelist_filename_in:
+        shutil.copyfile(os.path.join(data_path, "input.nml"), namelist_filename_out)
+    namelist = f90nml.read(namelist_filename_out)
+    total_ranks = 6 * namelist["fv_core_nml"]["layout"][0] * namelist["fv_core_nml"]["layout"][1]
+    grid_list = []
```

Comment:
So this is the list that holds grid data for all ranks right? Would it be safer to initialize this to be the size of total ranks, then put `grid_data` into it rank by rank?

Reply:
I think I can delete this code and use the general code below. This will add an extra singleton dimension at the start but that should be fine.